File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    if s[0] == ')' or s[-1] == '(':
        return False
    s = list(s)
    cnt = 0
    for ch in s:
        if ch == '(':
            cnt += 1
        else:
            cnt -= 1
            if cnt < 0:
                return False
    return cnt == 0

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        Deque<String> stack = new ArrayDeque<>();

        for (int i = 0; i < s.length(); i++) {
            if ((s.charAt(i)+"").equals("(")) 
                stack.push("(");
            else {
                if (stack.isEmpty() || (stack.peek()).equals(")"))
                    return false;
                stack.pop();
            }
        }
        if (stack.isEmpty())
            return true;
        return false;
    }
}